zimbra/src/main/java/fr/openent/zimbra/service/data/sqlTaskServices/SqlRecallTaskService.java

```diff
@@ -0,0 +1,57 @@
+package fr.openent.zimbra.service.data.sqlTaskServices;
+
+import fr.openent.zimbra.core.constants.Field;
+import fr.openent.zimbra.core.enums.TaskStatus;
+import fr.openent.zimbra.helper.PromiseHelper;
+import fr.openent.zimbra.model.action.Action;
+import fr.openent.zimbra.model.task.RecallTask;
+import fr.openent.zimbra.service.DbTaskService;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.entcore.common.sql.Sql;
+import org.entcore.common.sql.SqlResult;
+
+public class SqlRecallTaskService extends DbTaskService<RecallTask> {
+    private final String recallMailTable = schema + "." + "recall_mails";
+    public SqlRecallTaskService(String schema) {
+        super(schema);
+    }
+
+    @Override
+    public Future<JsonArray> retrieveTasksDataFromDB(TaskStatus status) {
+        Promise<JsonArray> promise = Promise.promise();
+        StringBuilder query = new StringBuilder();
+
+        query.append("SELECT * ")
+                .append(this.taskTable + " as recall_tasks ")
+                .append("JOIN " + this.actionTable + " as actions" + " on actions.id = recall_tasks.action_id")
+                .append("JOIN " + this.recallMailTable + " as recall_mails" + " on actions.id = recall_mails.action_id ")
+                .append("WHERE recall_tasks.status = ?;");
```

Comment:
Avec StringBuilder, qu'est-ce que tu utilises de plus hormis `.append` ? 
Si tu n'as pas d'autres usage (`insert`, `reverse` etc...) selon moi il vaut mieux utiliser un String normal : 

```suggestion
 String query = "SELECT * " +
                this.taskTable + " as recall_tasks " +
                "JOIN " + this.actionTable + " as actions" + " on actions.id = recall_tasks.action_id" +
                "JOIN " + this.recallMailTable + " as recall_mails" + " on actions.id = recall_mails.action_id " +
                "WHERE recall_tasks.status = ?;";
```